src/include/sof/audio/codec_adapter/codec/cadence.h

```diff
@@ -14,6 +14,20 @@
 #include <sof/audio/cadence/xa_memory_standards.h>
 
 #define LIB_NAME_MAX_LEN 30
+
+/*****************************************************************************/
+/* Codec API interface							     */
+/*****************************************************************************/
+extern xa_codec_func_t cadence_api_function;
+#define API cadence_api_function
+#define API_CALL(cd, cmd, sub_cmd, value, ret) \
+	do { \
+		ret = API((cd)->self, \
+			  (cmd), \
+			  (sub_cmd), \
+			  (value)); \
+	} while (0)
+
```

Comment:
I don't think we need extra API macro, also the API call can be placed on one line and also API_CALL can be named CODEC_API_CALL.
So, the code should look like this:

```
extern xa_codec_func_t cdns_api_func;

#define CODEC_API_CALL(cd, cmd, sub_cmd, value, ret) \
	do { \
		ret = cdns_api_func((cd)->self, (cmd),   (sub_cmd), (value)); 
	} while (0) 

```

Reply:
No need to preceed it with `CADENCE` or `CDNS` either, this will be called from `cadence.c` only.

Reply:
@mrajwa all right.